pybamm/expression_tree/functions.py

```diff
@@ -244,6 +256,25 @@ class SpecificFunction(Function):
     def __init__(self, function, child):
         super().__init__(function, child)
 
+    @classmethod
+    def _from_json(cls, function: Callable, snippet: dict):
```

Comment:
The type should be narrowed down if possible - `Callable[[..., ..., ...], ...].` Similarly, the `dict` type should also be narrowed down - `dict[..., ...]`.